### ISSUE TITLE
[1.8] babel typescript fails at class properties

### ISSUE THREAD

**Issue by yorrd** [meteor/meteor#10235]
I'm trying to compile with the new and cool @babel/preset-typescript. Everything is working quite well, except for the class properties. When i compile with the babel cli, it does compile, within meteor it doesn't. When I add --presets "meteor" to babel's cli, it fails for the same reason as it does when running meteor as usual. 

Here's my `.babelrc`

```
{
  "presets": ["@babel/preset-typescript", "@babel/preset-env"],
  "plugins": [
    ["@babel/plugin-proposal-decorators", { "legacy": true }],
    ["@babel/plugin-proposal-class-properties", { "loose": true }],
    "@babel/plugin-proposal-object-rest-spread",
    "@babel/plugin-syntax-dynamic-import"
  ]
}
```

I have a local version of the `ecmascript` package where I add `ts` files in the extensions list as so:

```
Plugin.registerCompiler(
  {
    extensions: ['js', 'jsx', 'ts', 'tsx'],
  },
  function() {
    return new BabelCompiler({
      react: true,
    });
  },
);
```

Versions are the latest, so rc16 at the moment.

EDIT: my packages:

```
meteor-base@1.4.0             # Packages every Meteor app needs to have
mobile-experience@1.0.5       # Packages for a great mobile UX
mongo@1.6.0-rc18.16                   # The database Meteor supports right now
reactive-var@1.0.11            # Reactive variable for tracker
tracker@1.2.0                 # Meteor's client-side reactive programming library

standard-minifier-css@1.5.0-rc18.16
standard-minifier-js@2.4.0-rc18.16

dynamic-import@0.5.0-rc18.16
server-render@0.3.1
es5-shim@4.8.0
static-html
ecmascript@0.12.0-rc18.16
```

**Comment by benjamn** [comment]
See https://github.com/meteor/meteor-feature-requests/issues/285#issuecomment-426646160. I'm afraid using on Babel to compile TypeScript without using the `tsc` compiler, and relying on custom `.babelrc` presets/plugins to add that support, is not a sustainable solution, because there will always be gaps between Babel's mimicry of `tsc` and the actual behavior of `tsc`. I think you would have a better time creating a new (local) `typescript` package that specifically handles the `.ts` and `.tsx` file extensions (and perhaps delegates some work to the `babel-compiler` package), while leaving the `ecmascript` package unmodified.

**Comment by yorrd** [comment]
Thanks for your insight! Do you feel like there would be a performance hit when compiling with tsc AND the ecmascript package? 

Background Info: I'm mainly searching for a new solution apart from `barbatus:typescript` because our rebuild times are > 10 seconds at the moment. A good solution < 3 seconds would save us actual real money and nerves, so I want to make sure we're going down the path with best performance.

**Comment by yorrd** [comment]
@benjamn not meaning to spam your inbox, but what's the current plan for typescript + meteor? The latest official version I know of is that Meteor recommends the barbatus:typescript package and that you're personally interested because you're using typescript yourself (I know this from a different issue on the topic). It'd just be great to know if we should make [this](https://github.com/meteor/meteor-feature-requests/issues/285#issuecomment-426732713) work as a faster alternative to barbatus:typescript or wait for Meteor to release something officially working without dirty hacks and workarounds. 

Your opinion and current official stance from Meteor would be greatly appreciated.

**Comment by yorrd** [comment]
tl;dr for anyone looking: Meteor [went on record](https://blog.meteor.com/meteor-1-8-erases-the-debts-of-1-7-77af4c931fe3) saying that there will be an official plugin. I'm not sure as to how active development on `barbatus:typescript` will be, I [opened an issue](https://github.com/barbatus/typescript/issues/74). 

Closing this issue as it doesn't serve a purpose anymore. Still wondering about the rough timeline though! :)
